src/coprocessor/dag/expr/builtin_control.rs

```diff
@@ -44,6 +44,30 @@ where
     }
 }
 
+/// See https://dev.mysql.com/doc/refman/5.7/en/case.html
+fn case_when<F, T>(expr: &FnCall, ctx: &StatementContext, row: &[Datum], f: F) -> Result<Option<T>>
+where
+    F: Fn(usize) -> Result<Option<T>>,
+{
+    let len = expr.children.len();
+    let pairs = len >> 1;
+    for id in 0..pairs {
```

Comment:
You mean `for chunk in expr.children.chunks(2)`?

Reply:
Ref: https://github.com/pingcap/tikv/blob/master/src/coprocessor/select/xeval/evaluator.rs#L388